ddtrace/tracer/tracer_test.go

```diff
@@ -1192,3 +1192,73 @@ func BenchmarkTracerStackFrames(b *testing.B) {
 		span.Finish(StackFrames(64, 0))
 	}
 }
+
+type testTracerStatsd struct {
+	measurements map[string]int
+	counts       map[string]int64
+	closed       bool
+}
+
+func (tg *testTracerStatsd) Incr(name string, tags []string, rate float64) error {
+	tg.measurements[name] = tg.measurements[name] + 1
+	return nil
+}
+
+func (tg *testTracerStatsd) Count(name string, value int64, tags []string, rate float64) error {
+	tg.measurements[name] = tg.measurements[name] + 1
+	tg.counts[name] = tg.counts[name] + value
+	return nil
+}
+
+func (tg *testTracerStatsd) Gauge(name string, value float64, tags []string, rate float64) error {
+	return nil
+}
+
+func (tg *testTracerStatsd) Timing(name string, value time.Duration, tags []string, rate float64) error {
+	tg.measurements[name] = tg.measurements[name] + 1
+	return nil
+}
+
+func (tg *testTracerStatsd) Close() error {
+	tg.closed = true
+	return nil
+}
+
+func withTracer(s statsdClient) StartOption {
+	return func(c *config) {
+		c.statsd = s
+	}
+}
+
+func TestTracerMetrics(t *testing.T) {
+	assert := assert.New(t)
+	statsd := &testTracerStatsd{
+		measurements: make(map[string]int),
+		counts:       make(map[string]int64),
+	}
+	tracer := newTracer(withTracer(statsd))
+	tracer.syncPush = make(chan struct{})
+	internal.SetGlobalTracer(tracer)
```

Comment:
I didn't change the name `tracer` because it's called `tracer` everywhere else in this file. 
https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/tracer_test.go#L172
https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/tracer_test.go#L187
https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/tracer_test.go#L193
and a bunch more.

I'm happy to make a change, but we should change it everywhere here. Another note is that we don't refer to the `tracer` package in this file, because we're *in* the `tracer` package. What do you think?

Reply:
I see. You're totally right about the name. I'd say it's fine to keep.